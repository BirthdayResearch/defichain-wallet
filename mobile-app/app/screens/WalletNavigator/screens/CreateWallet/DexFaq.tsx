import { ThemedScrollView, ThemedText } from '@components/themed'
import { AccordionContent, WalletAccordion } from '@components/WalletAccordion'
import { tailwind } from '@tailwind'
import { translate } from '@translations'

export function DexFaq (): JSX.Element {
  const faqContent: AccordionContent[] = [
    {
      title: translate('components/DexFaq', 'Transaction fees'),
      content: [{
        text: translate('components/DexFaq', 'Each transaction will be subject to a small amount of fees. The amount may vary depending on the network\'s congestion.'),
        type: 'paragraph'
      }]
    },
    {
      title: translate('components/DexFaq', 'What is slippage tolerance?'),
      content: [{
        text: translate('components/DexFaq', 'Due to the dynamic nature of the blockchain, prices on the DEX change every block.\n\nThis results in slippage, which is the difference between the final transaction price and the price displayed on the app before transaction confirmation.\n\nThe slippage tolerance feature allows you to indicate the maximum slippage you are willing to accept in a transaction.\n\nIf the slippage for your transaction is higher than your slippage tolerance, the transaction will not be completed and the original amount will be returned to your wallet.'),
        type: 'paragraph'
      }]
    }
  ]

  return (
    <ThemedScrollView
      contentContainerStyle={tailwind('p-6 pb-8')}
      testID='dex_faq'
    >
      <ThemedText
        style={tailwind('text-lg font-semibold')}
      >
        {translate('components/DexFaq', 'Decentralized Exchange')}
      </ThemedText>

      <ThemedText
        style={tailwind('mt-2 text-sm')}
      >
        {translate('components/DexFaq', 'The decentralized exchange function will allow users of DeFiChain to swap cryptocurrencies in a peer-to-peer fashion. The decentralized exchange function matches people for trading directly, without the need to buy and sell currency through an exchange.')}
      </ThemedText>

      <WalletAccordion
        testID='dex_faq_accordian'
        activeSections={[0]}
        title={translate('components/DexFaq', 'FREQUENTLY ASKED QUESTIONS')}
        content={faqContent}
      />
    </ThemedScrollView>
  )
}
