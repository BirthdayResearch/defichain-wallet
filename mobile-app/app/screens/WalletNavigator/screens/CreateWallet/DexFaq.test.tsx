import { render } from '@testing-library/react-native'
import { DexFaq } from './DexFaq'

jest.mock('@shared-contexts/ThemeProvider')
describe('DEX FAQ screen', () => {
  it('should match snapshot', async () => {
    const rendered = render(<DexFaq />)
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
