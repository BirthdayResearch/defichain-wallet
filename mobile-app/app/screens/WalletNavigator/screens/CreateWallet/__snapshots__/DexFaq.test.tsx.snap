// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DEX FAQ screen should match snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingBottom": 32,
      "paddingLeft": 24,
      "paddingRight": 24,
      "paddingTop": 24,
    }
  }
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "rgba(245, 245, 245, 1)",
      },
    ]
  }
  testID="dex_faq"
>
  <View>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Array [
            Object {
              "fontFamily": "SemiBoldFont",
              "fontSize": 18,
              "fontWeight": "600",
              "lineHeight": 28,
            },
            Object {
              "color": "rgba(0, 0, 0, 1)",
            },
          ],
        ]
      }
    >
      Decentralized Exchange
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Array [
            Object {
              "fontSize": 14,
              "lineHeight": 20,
              "marginTop": 8,
            },
            Object {
              "color": "rgba(0, 0, 0, 1)",
            },
          ],
        ]
      }
    >
      The decentralized exchange function will allow users of DeFiChain to swap cryptocurrencies in a peer-to-peer fashion. The decentralized exchange function matches people for trading directly, without the need to buy and sell currency through an exchange.
    </Text>
    <RCTScrollView
      style={
        Array [
          undefined,
          Object {
            "backgroundColor": "rgba(245, 245, 245, 1)",
          },
        ]
      }
      testID="dex_faq_accordian"
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "RegularFont",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              Array [
                Object {
                  "fontFamily": "MediumFont",
                  "fontSize": 12,
                  "fontWeight": "500",
                  "lineHeight": 16,
                  "marginTop": 32,
                },
                Object {
                  "color": "rgba(163, 163, 163, 1)",
                },
              ],
            ]
          }
          text="FREQUENTLY ASKED QUESTIONS"
        >
          FREQUENTLY ASKED QUESTIONS
        </Text>
        <View
          style={
            Array [
              Object {
                "borderBottomLeftRadius": 8,
                "borderBottomRightRadius": 8,
                "borderBottomWidth": 1,
                "borderLeftWidth": 1,
                "borderRightWidth": 1,
                "borderTopLeftRadius": 8,
                "borderTopRightRadius": 8,
                "borderTopWidth": 1,
                "marginTop": 8,
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "rgba(255, 255, 255, 1)",
                "borderBottomColor": "rgba(229, 229, 229, 1)",
                "borderLeftColor": "rgba(229, 229, 229, 1)",
                "borderRightColor": "rgba(229, 229, 229, 1)",
                "borderTopColor": "rgba(229, 229, 229, 1)",
              },
            ]
          }
        >
          <View
            style={Object {}}
          >
            <View
              accessibilityState={
                Object {
                  "expanded": true,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  Array [
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "paddingBottom": 16,
                        "paddingLeft": 16,
                        "paddingRight": 16,
                        "paddingTop": 16,
                      },
                      false,
                      Object {
                        "paddingBottom": 4,
                      },
                    ],
                    Object {
                      "borderBottomColor": "rgba(229, 229, 229, 1)",
                      "borderLeftColor": "rgba(229, 229, 229, 1)",
                      "borderRightColor": "rgba(229, 229, 229, 1)",
                      "borderTopColor": "rgba(229, 229, 229, 1)",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "RegularFont",
                        "fontSize": 16,
                        "fontWeight": "400",
                        "lineHeight": 24,
                      },
                      Array [
                        Array [
                          Object {
                            "flexBasis": "0%",
                            "flexGrow": 1,
                            "flexShrink": 1,
                            "fontSize": 14,
                            "lineHeight": 20,
                          },
                          Object {
                            "fontFamily": "SemiBoldFont",
                            "fontWeight": "600",
                          },
                        ],
                        Object {
                          "color": "rgba(23, 23, 23, 1)",
                        },
                      ],
                    ]
                  }
                >
                  Transaction fees
                </Text>
                <Text />
              </View>
            </View>
            <View
              pointerEvents="auto"
              style={Object {}}
            >
              <View
                forwardedRef={[Function]}
                onLayout={[Function]}
                style={Object {}}
              >
                <View
                  style={
                    Array [
                      Array [
                        Object {
                          "paddingBottom": 16,
                          "paddingLeft": 16,
                          "paddingRight": 16,
                          "paddingTop": 0,
                        },
                        Object {
                          "borderBottomWidth": 1,
                        },
                      ],
                      Object {
                        "borderBottomColor": "rgba(229, 229, 229, 1)",
                        "borderLeftColor": "rgba(229, 229, 229, 1)",
                        "borderRightColor": "rgba(229, 229, 229, 1)",
                        "borderTopColor": "rgba(229, 229, 229, 1)",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                        "justifyContent": "flex-start",
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "fontFamily": "RegularFont",
                            "fontSize": 16,
                            "fontWeight": "400",
                            "lineHeight": 24,
                          },
                          Array [
                            Object {
                              "flexBasis": "0%",
                              "flexGrow": 1,
                              "flexShrink": 1,
                              "fontSize": 14,
                              "lineHeight": 20,
                            },
                            Object {
                              "color": "rgba(82, 82, 82, 1)",
                            },
                          ],
                        ]
                      }
                    >
                      Each transaction will be subject to a small amount of fees. The amount may vary depending on the network's congestion.
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={Object {}}
          >
            <View
              accessibilityState={
                Object {
                  "expanded": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  Array [
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "paddingBottom": 16,
                        "paddingLeft": 16,
                        "paddingRight": 16,
                        "paddingTop": 16,
                      },
                      false,
                      false,
                    ],
                    Object {
                      "borderBottomColor": "rgba(229, 229, 229, 1)",
                      "borderLeftColor": "rgba(229, 229, 229, 1)",
                      "borderRightColor": "rgba(229, 229, 229, 1)",
                      "borderTopColor": "rgba(229, 229, 229, 1)",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "RegularFont",
                        "fontSize": 16,
                        "fontWeight": "400",
                        "lineHeight": 24,
                      },
                      Array [
                        Array [
                          Object {
                            "flexBasis": "0%",
                            "flexGrow": 1,
                            "flexShrink": 1,
                            "fontSize": 14,
                            "lineHeight": 20,
                          },
                          false,
                        ],
                        Object {
                          "color": "rgba(115, 115, 115, 1)",
                        },
                      ],
                    ]
                  }
                >
                  What is slippage tolerance?
                </Text>
                <Text />
              </View>
            </View>
            <View
              pointerEvents="none"
              style={
                Object {
                  "height": 0,
                  "overflow": "hidden",
                }
              }
            >
              <View
                forwardedRef={[Function]}
                onLayout={[Function]}
                style={Object {}}
              >
                <View
                  style={
                    Array [
                      Array [
                        Object {
                          "paddingBottom": 16,
                          "paddingLeft": 16,
                          "paddingRight": 16,
                          "paddingTop": 0,
                        },
                        false,
                      ],
                      Object {
                        "borderBottomColor": "rgba(229, 229, 229, 1)",
                        "borderLeftColor": "rgba(229, 229, 229, 1)",
                        "borderRightColor": "rgba(229, 229, 229, 1)",
                        "borderTopColor": "rgba(229, 229, 229, 1)",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                        "justifyContent": "flex-start",
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "fontFamily": "RegularFont",
                            "fontSize": 16,
                            "fontWeight": "400",
                            "lineHeight": 24,
                          },
                          Array [
                            Object {
                              "flexBasis": "0%",
                              "flexGrow": 1,
                              "flexShrink": 1,
                              "fontSize": 14,
                              "lineHeight": 20,
                            },
                            Object {
                              "color": "rgba(82, 82, 82, 1)",
                            },
                          ],
                        ]
                      }
                    >
                      Due to the dynamic nature of the blockchain, prices on the DEX change every block.

This results in slippage, which is the difference between the final transaction price and the price displayed on the app before transaction confirmation.

The slippage tolerance feature allows you to indicate the maximum slippage you are willing to accept in a transaction.

If the slippage for your transaction is higher than your slippage tolerance, the transaction will not be completed and the original amount will be returned to your wallet.
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTScrollView>
`;
